src/includes/class-alma-wc-plugin.php

```diff
@@ -599,7 +599,8 @@ public function get_asset_url( $path ) {
 	 */
 	private function get_payment_to_validate() {
 		// phpcs:ignore WordPress.Security.NonceVerification
-		$payment_id = isset( $_GET['pid'] ) ? $_GET['pid'] : null;
+		$id         = sanitize_text_field( $_GET['pid'] );
```

Comment:
can you also sanitize POST data ;)

Reply:
I did where I considered that I needed to be.